aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group/Program.cs

```diff
@@ -0,0 +1,201 @@
+using Data;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+var connection = new SqliteConnection("DataSource=:memory:");
+connection.Open();
+// Add InMemoryDatabase to the container.
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+{
+    options.UseSqlite(connection);
+});
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    // localhost:{port}/swagger
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.MapGet("/", () => "Hello World!");
+
+// todo endpoints
+var todos = app.MapGroup("/todos").WithTags("Todo Endpoints");
+todos.MapGet("/", GetAllTodos);
+todos.MapGet("/{id}", GetTodo);
+todos.MapPost("/", CreateTodo).AddRouteHandlerFilter((context, next) =>
+{
+    if (context.HttpContext.Request.ContentLength == 0)
+    {
+        context.HttpContext.Response.StatusCode = 400;
+        return new ValueTask<object?>(Results.BadRequest(new { Message = "Request body is too empty" }));
+    }
+    return next(context);
+});
+todos.MapPut("/{id}", UpdateTodo).AddRouteHandlerFilter((context, next) =>
+{
+    if (context.HttpContext.Request.ContentLength == 0)
+    {
+        context.HttpContext.Response.StatusCode = 400;
+        return new ValueTask<object?>(Results.BadRequest(new { Message = "Request body is too empty" }));
+    }
+    return next(context);
+});
+todos.MapDelete("/{id}", DeleteTodo);
+
+// note endpoints
+var notes = app.MapGroup("/notes").WithTags("Note Endpoints");
+notes.MapGet("/", GetAllNotes);
+notes.MapGet("/{id}", GetNote);
+notes.MapPost("/", CreateNote).AddRouteHandlerFilter((context, next) =>
+{
+    if (context.HttpContext.Request.ContentLength == 0)
+    {
+        context.HttpContext.Response.StatusCode = 400;
+        return new ValueTask<object?>(Results.BadRequest(new { Message = "Request body is empty" }));
+    }
+    return next(context);
+}); ;
+notes.MapPut("/{id}", UpdateNote).AddRouteHandlerFilter((context, next) =>
+{
+    if (context.HttpContext.Request.ContentLength == 0)
+    {
+        context.HttpContext.Response.StatusCode = 400;
+        return new ValueTask<object?>(Results.BadRequest(new { Message = "Request body is empty" }));
+    }
+    return next(context);
+}); ;
+notes.MapDelete("/{id}", DeleteNote);
+app.Run();
+
+
+// get all todos
+static async Task<IResult> GetAllTodos(ApplicationDbContext database)
```

Comment:
What do we think about putting all the methods we're mapping in a separate `RouteHandlers` class and file? I think it'd help make things look more organized.

Reply:
@Rick-Anderson  was telling me about making it portable for people to test. So I'm not really sure about the next step to make